lib/dfe/analytics/send_events.rb

```diff
@@ -18,6 +18,12 @@ def self.do(events)
       end
 
       def perform(events)
+        if DfE::Analytics.within_maintenance_window?
+          # Delaying / Queueing events - the redis bit
+          Rails.logger.info('Within BigQuery maintenance window. Events will be queued for later processing.')
+          return
+        end
+
```

Comment:
Need to remove this block here and move into the `do` method above.

```suggestion
```

In the `do` method replace lines 13-17 with the following:

``` Ruby
   if DfE::Analytics.within_maintenance_window?
     set(wait_until: DfE::Analytics.next_scheduled_time_after_maintenance_window).perform_later(events)
   elsif DfE::Analytics.async?
     perform_later(events)
   else
     perform_now(events)
   end
 ```